pkg/util/yaml/decoder.go

```diff
@@ -52,10 +53,16 @@ type YAMLToJSONDecoder struct {
 // stream in chunks by converting each document (as defined by
 // the YAML spec) into its own chunk, converting it to JSON via
 // yaml.YAMLToJSON, and then passing it to json.Decoder.
-func NewYAMLToJSONDecoder(r io.Reader) *YAMLToJSONDecoder {
+func NewYAMLToJSONDecoder(r io.Reader, bytes []byte, isStdin bool) *YAMLToJSONDecoder {
 	reader := bufio.NewReader(r)
-	return &YAMLToJSONDecoder{
-		reader: NewYAMLReader(reader),
+	if isStdin && sessionIsTTY() {
+		return &YAMLToJSONDecoder{
+			reader: NewYAMLByteReader(bytes),
```

Comment:
That's fine, we still use `Peek`. The only problem is how to handle bytes more than 4096, use a bigger size will not always help with this problem I believe.